libraries/standard/http/include/http_client.h

```diff
@@ -143,22 +142,30 @@ typedef enum HTTPStatus
 } HTTPStatus_t;
 
 /**
- * @brief Request headers buffer and how many headers exist in the buffer.
+ * @brief Represents header data that will be sent in an HTTP request. 
+ * 
+ * The memory for the header data buffer is supplied by the user. Information in
+ * the buffer will be filled by calling #HTTPClient_InitializeHeaders.
  */
 typedef struct HTTPRequestHeaders
 {
     
     /**
      * @brief Buffer to hold the raw HTTP request headers.
      * 
-     * This buffer is not available to modify until #HTTPNetworkInterface_t.recv
-     * for the response is invoked.
+     * This buffer is supplied by the application.
      * 
-     * If this buffer is shared with the response, then this buffer cannot be
-     * modified until #HTTPClient_Send() returns.
+     * This buffer is owned by the library during #HTTPClient_AddHeader, 
+     * #HTTPClient_AddRangeHeader, #HTTPClient_InitializeRequestHeaders, and
+     * #HTTPClient_Send. This buffer should not be modifed until 
+     * after these functions return.
+     * 
+     * For optimization this buffer may be shared with the response. The user 
```

Comment:
(Missing command) For optimzation \*,\* this buffer maybe...
I'm a bit wary of using the "shared" word as it may suggest concurrency. Instead, it would be _safer_ to stick with the _re-use_ word.

Reply:
sure